tests/route/test_forced_mgmt_route.py

```diff
@@ -33,27 +33,45 @@ def backup_restore_config(duthosts, enum_rand_one_per_hwsku_hostname):
     restore_config(duthost, CONFIG_DB, CONFIG_DB_BACKUP)
 
 
-def get_interface_reload_timestamp(duthost):
-    timestamp = duthost.command("sudo systemctl show --no-pager interfaces-config"
-                                " -p ExecMainExitTimestamp --value")["stdout"]
-    logger.info("interfaces config timestamp {}".format(timestamp))
+def get_file_hash(duthost, file):
+    hash = duthost.command("sha1sum {}".format(file))["stdout"]
+    logger.debug("file hash: {}".format(hash))
 
-    return timestamp
+    return hash
 
 
-def change_and_wait_interface_config_update(duthost, command, last_timestamp=None, timeout=10):
-    if not last_timestamp:
-        last_timestamp = get_interface_reload_timestamp(duthost)
+def wait_for_file_changed(duthost, file, action, *args, **kwargs):
+    original_hash = get_file_hash(duthost, file)
 
-    duthost.shell(command)
+    action(*args, **kwargs)
 
-    # Wait interfaces-config service finish
-    def log_exist(duthost):
-        latest_timestamp = get_interface_reload_timestamp(duthost)
-        return latest_timestamp != last_timestamp
+    def hash_changed(duthost, file):
+        latest_hash = get_file_hash(duthost, file)
+        return latest_hash != original_hash
 
-    exist = wait_until(timeout, 1, 0, log_exist, duthost)
-    pytest_assert(exist, "Not found interfaces-config update log: {}".format(command))
+    exist = wait_until(10, 1, 0, hash_changed, duthost, file)
+    pytest_assert(exist, "File {} does not change after 10 seconds.".format(file))
+
+
+def address_type(address):
+    return type(ipaddress.ip_network(str(address), False))
+
+
+def check_ip_rule_exist(duthost, ip, check_exist):
+    logging.warning("check_ip_rule_exist for ip:{} exist:{}".format(ip, check_exist))
+    rule_command = "ip rule list"
+    if address_type(ip) is ipaddress.IPv6Network:
+        rule_command = "ip -6 rule list"
+
+    ip_rules = duthost.command(rule_command)["stdout"]
+    logging.warning("ip rule list: {}".format(ip_rules))
+
+    rule = "32764:	from all to {} lookup default".format(ip)
```

Comment:
>32764 [](http://example.com/codeflow?start=12&length=5)

magic number. Could you use a named constant? Or point to the code which the number comes from.

Reply:
Fixed